lib/Client.js

```diff
@@ -126,6 +127,83 @@ export class Client {
     return data
   }
 
+  async queryTaxPayer (taxPayerId) {
+    assert(typeof taxPayerId === 'number' && /^[0-9]{8}$/.test(taxPayerId.toString()), 'taxPayerId must be an 8-digit number');
+
+    const xml = this._getXmlHeader('xmltaxpayer', 'agent') +
+      wrapWithElement(
+        'beallitasok', [
+          ...this._getAuthFields(),
+        ], 1) +
+        wrapWithElement('torzsszam', taxPayerId, 1) +
+      '</xmltaxpayer>'
+
+    const response = await this._sendRequest('action-szamla_agent_taxpayer', xml)
+    const parsedBody = await parseString(response.data)
+    const cleanParsedBody = removeNamespaces(parsedBody);
+    const cleanParsedBodyData = cleanParsedBody.QueryTaxpayerResponse;
+    const taxpayerValidity = cleanParsedBodyData.taxpayerValidity?.[0] === 'true';
+
+    if (!taxpayerValidity) {
+      return {
+        taxpayerValidity: false,
+      };
+    }
+
+    const taxpayerData = cleanParsedBodyData.taxpayerData?.[0] || {};
+    const taxNumberDetail = taxpayerData.taxNumberDetail?.[0] || {};
+  
+    return {
+      taxpayerValidity,
+      taxpayerId: taxNumberDetail.taxpayerId?.[0] || null,
+      vatCode: taxNumberDetail.vatCode?.[0] || null,
+      countyCode: taxNumberDetail.countyCode?.[0] || null,
+      taxpayerName: taxpayerData.taxpayerName?.[0] || null,
+      taxpayerShortName: taxpayerData.taxpayerShortName?.[0] || null,
+      address: this._extractAddress(taxpayerData.taxpayerAddressList?.[0]),
+    };
+  }
+
+  async registerCreditEntry (options, creditEntries) {
+    assert(typeof options.invoiceId === 'string' && options.invoiceId.trim().length > 1, 'invoiceId must be specified')
+    assert(Array.isArray(creditEntries) && creditEntries.length > 0, 'creditEntries must be specified and must be an array')
+    assert(creditEntries.every(entry => entry instanceof CreditEntry), 'All entries must be instances of CreditEntry')
+
+    const xml = this._getXmlHeader('xmlszamlakifiz', 'agentkifiz') +
+      wrapWithElement(
+        'beallitasok', [
+          ...this._getAuthFields(),
+          ['szamlaszam', options.invoiceId],
+          ['adoszam', options.taxNumber || ''],
```

Comment:
it is an optional filed. please don't set a default value . We save some bandwidth with it